include/oneapi/dpl/pstl/hetero/dpcpp/unseq_backend_sycl.h

```diff
@@ -202,103 +202,138 @@ struct __init_processing
 
 // Load elements consecutively from global memory, transform them, and apply a local reduction. Each local result is
 // stored in local memory.
-template <typename _ExecutionPolicy, ::std::uint8_t __iters_per_work_item, typename _Operation1, typename _Operation2,
-          typename _Tp, typename _Commutative>
+template <typename _ExecutionPolicy, typename _Operation1, typename _Operation2, typename _Tp, typename _Commutative,
+          std::uint8_t _VecSize>
 struct transform_reduce
```

Comment:
The changes here must be propagated to `sycl_traits.h` to update the forward declaration there (I believe this is causing the red CI).  Also, depending on which PR is merged first, #1521 the definition for `transform_reduce` will need to be updated accordingly to specify the template parameters which need to be checked to check device copyability.  
I can help make this adjustment if it is not clear to you, let me know.

Reply:
I believe it looks correct to me.  It should merely be adjusting the fwd declaration for now to match what is currently in the definition.  Once this or #1521 gets merged, we will need to adjust another line slightly to match the types in the specialization itself.

Reply:
@danhoeflinger #1521 was merged and I've rebased this PR. Please have a look if the updated forward declarations are fine. Thanks.

Reply:
I believe you will need to also update `implementation_details/device_copyable.pass`.  That should also be a good test of if we are still in good shape.  Otherwise the change looks good.

Reply:
Thanks, I've made the test change.